File: translation_hub/page/translation_workspace/translation_workspace.js
frappe.pages['translation-workspace'].on_page_load = function(wrapper) {
	var page = frappe.ui.make_app_page({
		parent: wrapper,
		title: 'Translation Workspace',
		single_column: false
	});

	page.set_primary_action("Create New Translation Job", () => {
		frappe.new_doc("Translation Job");
	});

    // Add dashboard
    let dashboard = new frappe.ui.Dashboard({
        parent: page.body,
        columns: 4
    });

    frappe.call({
        method: "translation_hub.page.translation_workspace.translation_workspace.get_dashboard_data",
        callback: function(r) {
            let data = r.message;

            dashboard.add_widget("total_apps", {
                type: "number",
                label: "Total Apps Tracked",
                value: data.total_apps,
                icon: "folder-open"
            });

            dashboard.add_widget("jobs_in_progress", {
                type: "number",
                label: "Jobs In Progress",
                value: data.jobs_in_progress,
                icon: "spinner"
            });

            dashboard.add_widget("jobs_completed_last_30_days", {
                type: "number",
                label: "Jobs Completed (Last 30 Days)",
                value: data.jobs_completed_last_30_days,
                icon: "check"
            });

            dashboard.add_widget("strings_translated", {
                type: "number",
                label: "Strings Translated (All Time)",
                value: data.strings_translated,
                icon: "language"
            });

            // Add recent jobs list
            let list_view = `
                <div class="frappe-card" style="margin-top: 20px;">
                    <div class="card-header">
                        <h5 class="card-title">Recent Translation Jobs</h5>
                    </div>
                    <div class="card-body">
                        <table class="table table-bordered">
                            <thead>
                                <tr>
                                    <th>Job Title</th>
                                    <th>App</th>
                                    <th>Language</th>
                                    <th>Status</th>
                                    <th>Progress</th>
                                </tr>
                            </thead>
                            <tbody>
                                ${data.recent_jobs.map(job => `
                                    <tr>
                                        <td><a href="/app/translation-job/${job.name}">${job.title}</a></td>
                                        <td>${job.source_app}</td>
                                        <td>${job.target_language}</td>
                                        <td><span class="indicator ${get_indicator_color(job.status)}">${job.status}</span></td>
                                        <td>
                                            <div class="progress">
                                                <div class="progress-bar" role="progressbar" style="width: ${job.progress_percentage}%" aria-valuenow="${job.progress_percentage}" aria-valuemin="0" aria-valuemax="100"></div>
                                            </div>
                                        </td>
                                    </tr>
                                `).join('')}
                            </tbody>
                        </table>
                    </div>
                </div>
            `;
            $(wrapper).find(".page-content").append(list_view);
        }
    });

    function get_indicator_color(status) {
        switch(status) {
            case "Completed": return "green";
            case "In Progress": return "blue";
            case "Queued": return "orange";
            case "Failed": return "red";
            case "Cancelled": return "darkgrey";
            default: return "grey";
        }
    }
}

File: translation_hub/page/translation_workspace/translation_workspace.py
# -*- coding: utf-8 -*-
# Copyright (c) 2025, Sebastian Rodrigues and contributors
# For license information, please see license.txt

from __future__ import unicode_literals
import frappe
from frappe.utils import nowdate, add_days

@frappe.whitelist()
def get_dashboard_data():
    total_apps = frappe.db.sql("""
        SELECT COUNT(DISTINCT source_app)
        FROM `tabTranslation Job`
    """)[0][0]

    jobs_in_progress = frappe.db.count("Translation Job", {"status": "In Progress"})

    thirty_days_ago = add_days(nowdate(), -30)
    jobs_completed_last_30_days = frappe.db.sql("""
        SELECT COUNT(*)
        FROM `tabTranslation Job`
        WHERE status = 'Completed' AND end_time >= %s
    """, thirty_days_ago)[0][0]

    strings_translated = frappe.db.sql("""
        SELECT SUM(translated_strings)
        FROM `tabTranslation Job`
        WHERE status = 'Completed'
    """)[0][0] or 0

    recent_jobs = frappe.get_list(
        "Translation Job",
        fields=["name", "title", "source_app", "target_language", "status", "progress_percentage"],
        order_by="modified DESC",
        limit=5
    )

    return {
        "total_apps": total_apps,
        "jobs_in_progress": jobs_in_progress,
        "jobs_completed_last_30_days": jobs_completed_last_30_days,
        "strings_translated": strings_translated,
        "recent_jobs": recent_jobs
    }
